numpy/_core/src/umath/umathmodule.c

```diff
@@ -279,6 +281,18 @@ int initumath(PyObject *m)
         return -1;
     }
 
+    PyObject *tmp;
+    if (PyDict_GetItemStringRef(d, "maximum", &tmp) < 0) {
+        return -1;
+    }
+    UFUNC_MAXIMUM = (PyUFuncObject *)tmp;
```

Comment:
Just store it directly?

Reply:
The old comment still applies again now: Just store it directly into UFUNC_MAXIMUM.

But I am not actually sure this is perfectly thread-safe. Can you try getting these from the n_ops struct in number.h. (I think we should use it, if including it this way around is awkward, it might actually make sense to move it into umathmodule.c)

Reply:
ok I'll try to use the n_ops from `number.h` first. It seems to work, I just need to cast ufunc, e.g., 
`        if ((PyObject *)ufunc == n_ops.maximum || (PyObject *)ufunc == n_ops.minimum) {`